sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -351,11 +399,54 @@ export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any
 /**
  * @internal
  * @ignore
- * Helper utility to build an instance of raw SQL parameter as `RawSqlParameter`
+ * Helper utility to extract array of userProperties key:value instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawUserProperties(parameters: { [key: string]: any } | undefined): any {
+  if (parameters == undefined) {
+    return undefined;
+  }
+
+  const rawParameters: RawKeyValuePair[] = [];
+  for (let [key, value] of Object.entries(parameters)) {
+    let type: string | number;
```

Comment:
There seem to be some more data types that aren't handled here (from the `TypeMapForRequestSerialization` enum). Why is that?

Reply:
Those four were meant for the SQLParameter. 
For the userProperties, I just borrowed string and number.

While checking the .NET code, I found that they support even more types like the timespan, double, and they seem to be using a different string for the date type.
https://github.com/Azure/azure-sdk-for-net/blob/7bd8e2aec47fbad148678947141cc501f4d5477c/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Filters/XmlObjectConvertor.cs#L56

I wanted to followup with the other languages before finalizing the supported types. For now, it would be "string" and "number".

Reply:
- Also, in comparison with .NET, JS doesn't have timespan, and I'm not sure if I can determine double and long properly.
- And in JS, for the SQLParameter, for the date type, we were adding "l28:date" and in the link I shared, .NET adds "l28:dateTime". I need to verify if it is even working in JS.

Python added string support only. Shared my concerns with Yijun, he'll also look into supporting more types in python.

Reply:
Okay, afterthoughts on the types allowed in userParameters.
- Service did not respect the "l68:date" type (which is what the SqlParameter was relying upon). There is no service documentation on which types are supported.
- For now, I have added the support for "string", "number" and "boolean", I'd like to wrap up this PR with these for now.
- In a later PR, look into the other types with more emphasized testing on the supported types for the `SqlParameter`.
- Python supports strings only as of now.  From [here](https://github.com/Azure/azure-sdk-for-net/blob/7bd8e2aec47fbad148678947141cc501f4d5477c/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Filters/XmlObjectConvertor.cs#L56), .NET seems to support "long", "double", "dateTime" and "duration" as the types other than the ones I have added.
- I'd also like to compare this with the `userProperties` set on the message since they are the ones that should be matched with these from the correlationFilters, I don't think we do any sort of type validations for the userProperties on the message, but we seem to be applying JSON.stringify() on the message/message-body. In the end, we need to ensure that the userProperties in the rules are capable of matching the ones set on the message.

Reply:
So long as you file an issue for the work I'm good. Feel free to resolve afterwards.

Reply:
Logged #9850